vcd/resource_vcd_edgegateway.go

```diff
@@ -0,0 +1,201 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdEdgeGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdEdgeGatewayCreate,
+		Delete: resourceVcdEdgeGatewayDelete,
+		Read:   resourceVcdEdgeGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vdc": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			/*
+					// RFC:
+				    // Waiting on a decision whether such parameters are needed
+					"delete_force": &schema.Schema{
+						Type:        schema.TypeBool,
+						Required:    true,
+						ForceNew:    true,
+						Description: "TBD: no description in the API, and undefined behavior in practice",
+					},
+					"delete_recursive": &schema.Schema{
+						Type:        schema.TypeBool,
+						Required:    true,
+						ForceNew:    true,
+						Description: "TBD: no description in the API, and undefined behavior in practice",
+					},
+			*/
+			"advanced_networking": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Description: "True if the gateway uses advanced networking",
+			},
+			"backing_config": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Configuration of the vShield edge VM for this gateway. One of: compact, full",
+			},
+			"ha_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Description: "Enable high availability on this edge gateway",
+			},
+			"external_networks": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A list of external networks to be used by the edge gateway",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"default_gateway": &schema.Schema{
```

Comment:
Personally for me `default_gateway` implies that I need to specify IP address. Wondering if it could be something like `default_gateway_network` or something similar.

Reply:
I agree, in other places gateway is IP. Though `_network` could be not enough, cause only `external` only applicable ?

Reply:
I like the shortness of `default_gateway`. If you are used to GUI operations, it doesn't sound strange, as you set the gateway from the external network. Add to it that you will get an error if your value for default gateway is not included in the external networks list, and I think this interface is reasonable.

Rather than making it unbearably long as `default_gateway_external_network`, I prefer leaving it as it is.

Reply:
cc: @lvirbalas

Reply:
Agree with @dataclouder . `default_gateway_external_network` is too long. Probably this one is ok then, unless we have some shorter meaningful title option.

Reply:
I agree now that `default_gateway_external_network` is too long.

Reply:
Ha! First time I scanned over this, I actually thought this *is* an IP, until I read the description...
`default_gateway_network` is more intuitive though longer.

Reply:
changed